dev/system-test/firestore.ts

```diff
@@ -2629,69 +2629,95 @@ describe('BulkWriter class', () => {
+      //     ├── anna
+      //     └── bob
+      //         └── parentsCol
+      //             ├── charlie
+      //             └── daniel
+      //                 └── childCol
+      //                     ├── ernie
+      //                     └── francis
+      const batch = firestore.batch();
+      batch.set(randomCol.doc('anna'), {name: 'anna'});
+      batch.set(randomCol.doc('bob'), {name: 'bob'});
+      batch.set(randomCol.doc('bob/parentsCol/charlie'), {name: 'charlie'});
+      batch.set(randomCol.doc('bob/parentsCol/daniel'), {name: 'daniel'});
+      batch.set(randomCol.doc('bob/parentsCol/daniel/childCol/ernie'), {
+        name: 'ernie',
+      });
+      batch.set(randomCol.doc('bob/parentsCol/daniel/childCol/francis'), {
+        name: 'francis',
+      });
+      await batch.commit();
+    });
 
-    // Query all descendants of documents.
-    descendantsStream = await firestore._getAllDescendants(
-      randomCol.doc('bob')
-    );
-    expect(await numStreamItems(descendantsStream)).to.equal(4);
-    descendantsStream = await firestore._getAllDescendants(
-      randomCol.doc('bob/parentsCol/daniel')
-    );
-    expect(await numStreamItems(descendantsStream)).to.equal(2);
-    descendantsStream = await firestore._getAllDescendants(
-      randomCol.doc('anna')
-    );
-    expect(await numStreamItems(descendantsStream)).to.equal(0);
+    it('recursiveDelete on top-level collection', async () => {
+      await firestore.recursiveDelete(randomCol);
+      expect(await countCollectionChildren(randomCol)).to.equal(0);
+    });
+
+    it('recursiveDelete on nested collection', async () => {
+      const coll = randomCol.doc('bob').collection('parentsCol');
+      await firestore.recursiveDelete(coll);
+
+      expect(await countCollectionChildren(coll)).to.equal(0);
+      expect(await countCollectionChildren(randomCol)).to.equal(2);
+    });
+
+    it('recursiveDelete on nested document', async () => {
+      const doc = randomCol.doc('bob/parentsCol/daniel');
+      await firestore.recursiveDelete(doc);
+
+      const docSnap = await doc.get();
+      expect(docSnap.exists).to.be.false;
+      expect(await countDocumentChildren(randomCol.doc('bob'))).to.equal(1);
+      expect(await countCollectionChildren(randomCol)).to.equal(3);
+    });
+
+    it('recursiveDelete on leaf document', async () => {
+      const doc = randomCol.doc('bob/parentsCol/daniel/childCol/ernie');
+      await firestore.recursiveDelete(doc);
+
+      const docSnap = await doc.get();
+      expect(docSnap.exists).to.be.false;
+      expect(await countCollectionChildren(randomCol)).to.equal(5);
+    });
+
+    it('recursiveDelete with custom BulkWriter instance', async () => {
+      const bulkWriter = firestore.bulkWriter();
+      let callbackCount = 0;
+      bulkWriter.onWriteResult(() => {
+        callbackCount++;
+      });
+      await firestore.recursiveDelete(randomCol, bulkWriter);
+      expect(callbackCount).to.equal(6);
+    });
```

Comment:
Can we also get a test that has "randomCollA/docA" and "randomCollB/docB" and only deletes randomCollA? There might be some nuances in how to delete parent collection (in other ports) and I want to make sure that all of our clients handle this case correctly.